notebooks/ir_ima_visualization/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,327 @@
+import numpy as np
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from ginga.util.zscale import zscale
+import matplotlib.patheffects as path_effects
+
+def read_wfc3(filename):
+    '''
+    Read a full-frame IR image and return the datacube plus integration times for each read.
+    Requires the path to a RAW full-frame IR image fits file (filename.)
+
+    Parameters
+    ----------
+    filename : str
+        Path to a RAW full-frame IR image fits file.
+
+    Returns
+    -------
+    cube : array-like
+        1024x1024xNSAMP datacube of the IR image in ascending time order, 
+        where NSAMP is the number of samples taken.
+  
+    integ_time : array-like
+        Integration times associated with the datacube in ascending order.
+        
+    '''
+    with fits.open(filename) as f:
+        hdr = f[0].header
+        NSAMP = hdr['NSAMP']
+        hdr1 = f[1].header
+        cube = np.zeros((hdr1['NAXIS1'],hdr1['NAXIS2'],NSAMP), dtype = float)
+        integ_time = np.zeros(shape = (NSAMP))
+
+        for i in range(1, NSAMP+1):
+            cube[:,:,i-1] = f[('SCI', i)].data
+            integ_time[i-1] = f[('TIME', i)].header['PIXVALUE']
+
+        cube = cube[:,:,::-1]
+        integ_time = integ_time[::-1]
+    return cube, integ_time
+
+
+
+def compute_diff_imas(cube, integ_time, diff_method):
+    
+    '''
+    Compute the difference in signal between reads of a WFC3 IR IMA file.
+    
+    Parameters
+    ----------
+    cube : array-like
+       1024x1024xNSAMP datacube of the IR image in ascending time order, 
+       where NSAMP is the number of samples taken.
+
+    integ_time : array-like
+       Integration times associated with the datacube in ascending order.
+
+    diff_method: str
+       The method of finding the difference between reads. 
+       Either "instantaneous" or "cumulative".
+           
+    Returns
+    --------
+    diff : array-like
+        1024x1024x(NSAMP-1) datacube of the differebce between IR IMA reads in ascending time order, 
+        where NSAMP is the number of samples taken.
+        
+    '''
+    
+    if diff_method == 'instantaneous':
+        ima_j = cube[:, :, 1:]
+        ima_j_1 = cube[:,:,0:-1]
+        diff = ((ima_j*(integ_time[1:]-integ_time[0]))-\
```

Comment:
`t_0, t_j, and t_j_1` variables may also help with readability. I apologize, I thought that was assumed from my comment.

```
if diff_method == 'instantaneous':
    ima_j = cube[:, :, 1:]
    ima_j_1 = cube[:, :, 0:-1]
    t_0 = integ_time[0]
    t_j = integ_time[1:]
    t_j_1 = integ_time[0:-1]
    # I think this is right but double check just in case
    inst_diff = (ima_j * (t_j - t_0) - ima_j_1 * (t_j_1 - t_0)) / (t_j - t_j_1)
```